src/middlewared/middlewared/utils/__init__.py

```diff
@@ -322,6 +322,9 @@ def filterop(f):
     if options.get('get') is True:
         return rv[0]
 
+    if options.get('limit'):
+        return rv[:options['limit']]
```

Comment:
We should probably implement the same thing in datastore.query, to keep when equal.